homeassistant/components/planifneige/sensor.py

```diff
@@ -0,0 +1,135 @@
+from homeassistant.helpers.restore_state import RestoreEntity
+
+DEPENDENCIES = ['planifneige']
+
+_LOGGER = logging.getLogger(__name__)
+
+DB_TIMEFORMAT = '%Y-%m-%d %H:%M:%S'
+
+ATTR_STREET_SIDE_ID = 'street_side_id'
+ATTR_START_PLAN_DATE = 'start_plan_date'
+ATTR_END_PLAN_DATE = 'end_plan_date'
+ATTR_START_REPLAN_DATE = 'start_replan_date'
+ATTR_END_REPLAN_DATE = 'end_replan_date'
+ATTR_UPDATED = 'updated'
+
+STREET_STATE = {
+    '0': ['Snowed', 'mdi:snowflake'],  # street is snowed in
+    '1': ['Clear', 'mdi:road'],  # street is clear
+    '2': ['Planned', 'mdi:clock-outline'],  # street is planned for clearing
+    '3': ['Replanned', 'mdi:clock-alert-outline'],  # clearing date replanned
+    '4': ['Snowed', 'mdi:snowflake'],  # to be planned, still snowed in
+    '5': ['Clearing', 'mdi:bulldozer'],  # trucks loading, still clearing
+    '10': ['Ploughed', 'mdi:road']  # street cleared, snow not loaded
+}
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the PlanifNeige platform."""
+    data = hass.data[DATA_PLANIFNEIGE]
+    async_add_entities(
+        [PlanifNeigeSensor(data, sensor) for sensor in discovery_info]
+    )
+
+
+class PlanifNeigeSensor(RestoreEntity):
+    """PlanifNeige sensor."""
+
+    def __init__(self, data, sensor):
+        """Initialize the sensor."""
+        self._data = data.data
+        self._state = None
+        self._name = sensor[CONF_NAME]
+        self._street_id = sensor[CONF_STREETID]
+        self._icon = ""
+        self._start_plan_date = None
+        self._end_plan_date = None
+        self._start_replan_date = None
+        self._end_replan_date = None
+        self._date_updated = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def street_id(self):
+        """Return the street_id of the sensor."""
+        return self._street_id
+
+    @property
+    def start_plan_date(self):
+        """Return the start planned date of the sensor."""
+        return self.format_dbtime(self._start_plan_date)
+
+    @property
+    def end_plan_date(self):
+        """Return the end planned date of the sensor."""
+        return self.format_dbtime(self._end_plan_date)
+
+    @property
+    def start_replan_date(self):
+        """Return the start replanned date of the sensor."""
+        return self.format_dbtime(self._start_replan_date)
+
+    @property
+    def end_replan_date(self):
+        """Return the end replanned date of the sensor."""
+        return self.format_dbtime(self._end_replan_date)
+
+    @property
+    def date_updated(self):
+        """Return the date updated of the sensor."""
+        return self.format_dbtime(self._date_updated)
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    def update(self):
```

Comment:
We need to connect `update` to the dispatch signal in `async_added_to_hass` entity coroutine method.

Also set `should_poll` property to `False`.

Reply:
Not sure how to proceed with async_added_to_hass. Would it be something similar to this:
https://github.com/home-assistant/home-assistant/blob/865839bbe1c6c737e7306e3cce953f11cad9e5f0/homeassistant/components/ads/binary_sensor.py#L49-L55

Reply:
Something like this:
```py
@callback
def async_update_callback(self):
    self.async_schedule_update_ha_state(True)

async def async_added_to_hass(self):
    async_dispatcher_connect(
        self.hass, self.async_update_callback)
```